test/setup.js

```diff
@@ -22,8 +22,10 @@ if (typeof window !== 'undefined') {
 if (typeof document !== 'undefined' && typeof window !== 'undefined') {
   document.documentElement.clientWidth = window.innerWidth;
   document.documentElement.clientHeight = window.innerHeight;
+  // So that we can test window visibility within tests.
+  document.webkitvisibilitychange = () => {};
+  document.webkitHidden = true;
```

Comment:
I think it makes sense to have the default has 

```js
document.hidden = false;
```

rather than 'true'.

Reply:
@alexreardon I am a little stuck about mocking the visibility event for the unit tests.

 The `document` object that is used for the unit tests has a property `hidden` but no `visibilitychange` function. So within `test/setup.js` I added the `visibilitychange` function to the `document` object and within that function tried set `document.hidden = true`.

I presumed that would mock the window being hidden but the function does not get called. 

So I am wondering does anyone have any suggestions how to mock this?

Reply:
You can trigger your own events manually. See `user-input-util`

```js
export const dispatchWindowEvent = (eventName: string, options?: Object = {}): Event => {
  const event: Event = document.createEvent('Event');
  event.initEvent(eventName, true, true);

  // override properties on the event itself
  Object.assign(event, options);

  window.dispatchEvent(event);

  return event;
};
```